### DISCUSSION TITLE
Question about the syntax.

### DISCUSSION THREAD

**Discussion by Galllaa** [ramokz/phantom-camera#98] [General]
Hi, thank you for this awesome tool. One question how should i call the ( "Follow Parameters/Target Offset" ) PhantomCamera2D ? So what im trying to achieve is a "pan" or " look ahead" (idk how to call it) when im running im trying to push the offset of the camera ahead.  Should i call the offset of the camera2d or is there a way to call it in the PhantomCamera2D? Still learning Godot. Thank you in advance.

**Comment by ramokz** [comment]
Good question! Can think of two ways of offsetting the camera and move it ahead of the tracked Node during runtime.

1. Like you said yourself, you can modify the Target Offset property via code. If you reference the PhantomCamera2D node that you're using to, e.g., `phantom_camera`, it'll look like `phantom_camera.Properties.follow_target_offset_2D = Vector2(newOffsetX, newOffsetY)`. Note, doing this will instantly snap to that position, which can feel abrupt, but you can also manually lerp the offset value to smooth out the change.

2. Alternatively, you can use two PhantomCamera2Ds; a default one and a "look ahead" one that is offset to where you want it to be. You can then change the priority of the "look ahead" one to a higher value than the default one when you want the camera to be positioned there. The benefit of doing it this way is that you can easily fine-tune how fast, if at all, the camera should transition between a default state vs. a "look at" state.

If that answers your question?

**Reply by Galllaa** [reply]
Yea I see what u mean with the first way. How could I change the current PhantomCamera2D ? I mean how to switch to the "look ahead" camera ?

**Reply by ramokz** [reply]
You can use the `set_priority` method ([more info here](https://github.com/ramokz/phantom-camera/wiki/Properties:-Priority-(2D-&-3D))), which is how you would change between PhantomCameras via code.

So you can write `look_ahead_phantom_camera.set_priority(value)`, where the value is just an integer that's higher than the currently active PhantomCamera's—the priority value for each PhantomCamera node can be seen in their respective inspector panels. To change back to the previous camera, you would just need to call the method again with a lower value.

**Reply by Galllaa** [reply]
Thank you :) last question is there a limit how many PhantomCamera2D nodes you can have in one scene ?

**Reply by ramokz** [reply]
No problem :) PhantomCamera nodes are like other Godot nodes, so you can have as many as you want. One of the core features of the plugin is to have multiple in a given scene and dynamically swap between them as needed.
